cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java

```diff
@@ -116,7 +118,7 @@ protected void startUp() throws Exception {
       @Override
       public Boolean call() throws Exception {
         try {
-          getDSFramework().getInstances(Id.Namespace.DEFAULT);
+          namespaceQueryAdmin.list();
```

Comment:
That's a bit awkward to use namespace query admin to test whether dataservice is up or not. Can we just try to list the `SYSTEM` namespace?